modules/docs/markdown/03-protocols/03-aws/04-code-size.md

```diff
@@ -0,0 +1,88 @@
+
+1. Pre-build the AWS artifacts for the services you use, and publish them to your organisation's internal artifact registry
+2. Limit the amount of code being generated to the subset of service operations your project actually uses
+
+Option 1 hopefully doesn't need much explanation - once the code has been generated, it can be published as a regular Scala artifact into a registry of your choice, such as JFrog Artifactory.
+
+Option 2 is what we will focus on here, as it allows you to drastically reduce code size as part of your own project, without having to publish anything anywhere.
+
+To demonstrate this approach better, let's immediately start with an example.
+
+## Example
+
+Let's say you decided to build a service that uses [AWS Comprehend](https://docs.aws.amazon.com/comprehend/latest/dg/sdk-general-information-section.html) to detect sentiment in text provided by the user.
+
+You start with the [smithy4s.g8 template](https://github.com/disneystreaming/smithy4s.g8/), and, following the [documentation](01-aws.md), instruct Smithy4s plugin to generate you a client SDK for Comprehend:
+
+```scala
+    smithy4sAwsSpecs ++= Seq(AWS.comprehend)
+```
+
+As you run `compile` in your build tool, you can see that there are 428 files being compiled. Uh-oh.
+
+Those 428 files cover the entirety of Comprehend's [84 operations](https://docs.aws.amazon.com/comprehend/latest/APIReference/API_Operations.html) and all the dataclasses supporting them. But we don't need all of these operations - we only care about [DetectSentiment](https://docs.aws.amazon.com/comprehend/latest/APIReference/API_DetectSentiment.html).
+
+### @only annotation
+
+To express exactly that, Smithy4s ships with a built-in annotation `smithy4s.meta#only`, which can be applied to operations (and **operations only**) that you would like to keep in generated code, along with all the other Smithy shapes they reference. 
+
+The exact semantics of the annotation are as follows:
+
+- It can only be applied to operations
+- For any service that has operations tagged with this annotation, the rest of the operations will be **removed**, along with all the shapes (structures, aliases, unions, enums, etc.) that those operations reference, as long as those shapes are deemed **removable** (see below)
+- A shape is deemed **removable** if it's only reachable from the removed operations (either directly or transitively)
+- Services that don't have *any* operations tagged with this annotation are assumed to have *all* of their operations tagged implicitly (so that all the operations are kept)
+
+The key mechanism that allows this operation to be usable with AWS client generation is that we can [apply annotations post factum on shapes defined elsewhere](https://disneystreaming.github.io/smithy4s/docs/guides/model-preprocessing#note-on-third-party-models).
+
+So to only preserve `DetectSentiment` operation, we can create a Smithy file in our project with these contents:
```

Comment:
**issue:** this says "only `DetectSentiment` but you're applying `only` on two ops. Which is it?